### ISSUE TITLE
Question: how vflow handle IPFIX and Netflowv9sampling rate

### ISSUE THREAD

**Issue by Slepwin** [VerizonDigital/vflow#145]
Hi,
I have a question how vflow handle IPFIX/Netflow v9 sampling rate, does it get from **Option Data Sets** and multiply by the number of bytes and packets automatically?

**Comment by Slepwin** [comment]
Seems like vflow completely ignores sampling interval field in IPFIX and Netflow packets and can't provide accurate information about traffic volume.

**Comment by Slepwin** [comment]
Maybe it have some configuration knob to set sampling rate manually?

**Comment by Slepwin** [comment]
@mehrdadrad could you please clarify.

**Comment by mehrdadrad** [comment]
@Slepwin  samplingInterval IANA element id #34 deprecated please check samplingPacketInterval 
https://www.iana.org/assignments/ipfix/ipfix.xhtml

**Comment by Slepwin** [comment]
@mehrdadrad i have a ipfix.elements file with samplingPacketInterval element in /etc/vflow directory but i can confirm vflow doesn't take into account this info and doesn't multiple bytes to sampling rate provided from routers (tested on Juniper MX).

**Comment by mehrdadrad** [comment]
vFlow doesn't multiply, if router sends the samplingPacketInterval then you should get it at dataset (json)
did you try tcpdump/wireshark to make sure it comes from Juniper MX router?

**Comment by Slepwin** [comment]
> vFlow doesn't multiply, if router sends the samplingPacketInterval then you should get it at dataset (json)
> did you try tcpdump/wireshark to make sure it comes from Juniper MX router?

Yes, i can confirm MX router send samplingPacketInterval for IPFIX and another collector do it (multiply) automatically. I also can't see samplingPacketInterval on vflow with debug mode.

**Comment by mehrdadrad** [comment]
can you send me a pcap?
